Uncrustify erroneously removes semicolon

Current Git HEAD revision introduces a parsing error by erroneously remove a semicolon that cannot be removed.
Operating system is Ubuntu 20.04.
[p.tar.gz](https://github.com/uncrustify/uncrustify/files/7336245/p.tar.gz)

```
$ cat p.c
#include <stdio.h>

void foo(unsigned flags, unsigned COMMENT) {
	/* Discard file comment if any */
	if ((flags & COMMENT) != 0) {
	    while (getchar() != 0) /* null */ ;
	}
}
$ gcc -c p.c
$ cat p.cfg
mod_full_brace_while            = force
nl_while_brace                  = force
mod_remove_extra_semicolon      = true
$ ./uncrustify -c p.cfg -f p.c -o q.c
Parsing: p.c as language C
$ gcc -c q.c
q.c: In function ‘foo’:
q.c:6:37: error: expected expression before ‘}’ token
    6 |   while (getchar() != 0) /* null */ } {
      |                                     ^
1|$ cat q.c
#include <stdio.h>

void foo(unsigned flags, unsigned COMMENT) {
	/* Discard file comment if any */
	if ((flags & COMMENT) != 0) {
		while (getchar() != 0) /* null */ } {
}
}
$ ./uncrustify -v
Uncrustify_d-0.73.0-210-89d022f5
$
```

All files are contained in the attached archive.

--- a/src/brace_cleanup.cpp
+++ b/src/brace_cleanup.cpp
@@ -1365,6 +1365,7 @@ static chunk_t *insert_vbrace(chunk_t *pc, bool after, const ParseFrame &frm)
    {
       chunk.flags &= ~PCF_IN_PREPROC;
    }
+   bool ref_is_comment = chunk_is_comment(ref);      // Issue #3351
 
    while (  chunk_is_newline(ref)
          || chunk_is_comment(ref))
@@ -1400,6 +1401,11 @@ static chunk_t *insert_vbrace(chunk_t *pc, bool after, const ParseFrame &frm)
             ref = chunk_get_next_nc(ref);
          }
       }
+   }
+
+   if (ref_is_comment)                                      // Issue #3351
+   {
+      ref = chunk_get_next(ref);
    }
 
    if (ref == nullptr)

--- a/tests/c.test
+++ b/tests/c.test
@@ -427,3 +427,4 @@
 10038  c/indent_case_comment-false.cfg            c/Issue_3362.c
 10039  c/Issue_3370.cfg                           c/Issue_3370.c
 10040  c/Issue_3377.cfg                           c/Issue_3377.c
+10041  c/Issue_3351.cfg                           c/Issue_3351.c

--- a/tests/config/c/Issue_3351.cfg
+++ b/tests/config/c/Issue_3351.cfg
@@ -1,0 +1,3 @@
+mod_full_brace_while       = force
+nl_while_brace             = force
+mod_remove_extra_semicolon = true

--- a/tests/expected/c/10041-Issue_3351.c
+++ b/tests/expected/c/10041-Issue_3351.c
@@ -1,0 +1,10 @@
+#include <stdio.h>
+
+void foo(unsigned flags, unsigned COMMENT) {
+	/* Discard file comment if any */
+	if ((flags & COMMENT) != 0) {
+		while (getchar() != 0) /* null */
+		{
+		}
+	}
+}

--- a/tests/input/c/Issue_3351.c
+++ b/tests/input/c/Issue_3351.c
@@ -1,0 +1,8 @@
+#include <stdio.h>
+
+void foo(unsigned flags, unsigned COMMENT) {
+	/* Discard file comment if any */
+	if ((flags & COMMENT) != 0) {
+	    while (getchar() != 0) /* null */ ;
+	}
+}

